wasm_src/gsl_wrapper/gsl_wrapper.cc

```diff
@@ -0,0 +1,213 @@
+#include "gsl/gsl_statistics_double.h"
+#include "gsl/gsl_sort_int.h"
+#include <math.h>
+#include "gsl/gsl_fit.h"
+#include "gsl/gsl_multifit.h"
+
+extern "C" {
+
+int EMSCRIPTEN_KEEPALIVE filterBoxcar(double* yInArray, const int N, double* yOutArray, const int kernel) {
+    int status = 0;    /* return value: 0 = success */
+    gsl_vector_view yIn = gsl_vector_view_array(yInArray, N);
+    double* window = new double[kernel];
+    size_t H, J;
+    if (kernel % 2 == 0) {
+        H = kernel / 2 - 1;
+        J = kernel - H - 1;
+    } else {
+        H = (kernel - 1) / 2;
+        J = (kernel - 1) / 2;
+    }
+
+    for (size_t i = 0; i < N; ++i) {
+        size_t wsize = gsl_movstat_fill(GSL_MOVSTAT_END_PADVALUE, &yIn.vector, i, H, J, window);
+        yOutArray[i] = gsl_stats_mean(window, 1, wsize);
+    }
+
+    delete[] window;
+
+    return status;
+}
+
+int EMSCRIPTEN_KEEPALIVE filterGaussian(double* yInArray, const int N, double* yOutArray, const int kernel, const double alpha) {
+    int status = 0;    /* return value: 0 = success */
+    gsl_vector_view yIn = gsl_vector_view_array(yInArray, N);
+    gsl_vector_view yOut = gsl_vector_view_array(yOutArray, N);
+    gsl_filter_gaussian_workspace* w = gsl_filter_gaussian_alloc(kernel);
+
+    gsl_filter_gaussian(GSL_FILTER_END_PADVALUE, alpha, 0, &yIn.vector, &yOut.vector, w);
+
+    gsl_filter_gaussian_free(w);
+
+    return status;
+}
+
+double hanningWindow(const size_t n, double window[], void* params) {
+    double val = 0;
+    double sum = 0;
+
+    for (size_t i = 0; i < n; i++) {
+        double hanningVal = (0.5 * (1 - gsl_sf_cos(2 * M_PI * (i + 1) / (n + 1))));
+        val += hanningVal * window[i];
+        sum += hanningVal;
+    }
+
+    return val / sum;
+}
+
+int EMSCRIPTEN_KEEPALIVE filterHanning(double* yInArray, const int N, double* yOutArray, const int kernel) {
+    int status = 0;    /* return value: 0 = success */
+    gsl_vector_view yIn = gsl_vector_view_array(yInArray, N);
+    gsl_vector_view yOut = gsl_vector_view_array(yOutArray, N);
+    gsl_movstat_workspace* w = gsl_movstat_alloc(kernel);
+    gsl_movstat_function F;
+
+    F.function = hanningWindow;
+    gsl_movstat_apply(GSL_MOVSTAT_END_PADVALUE, &F, &yIn.vector, &yOut.vector, w);
+
+    gsl_movstat_free(w);
+
+    return status;
+}
+
+int EMSCRIPTEN_KEEPALIVE filterDecimation(double* xInArray, double* yInArray, const int inN, double* xOutArray, double* yOutArray, const int outN, const int decimationWidth) {
+    int status = 0;    /* return value: 0 = success */
+    int* indexArray = new int[outN];
+    for (size_t i = 0; i <= inN / decimationWidth; i++) {
+        if (i == inN / decimationWidth) {
+            if (inN % decimationWidth == 0) {
+                break;
+            } else if (inN % decimationWidth == 1) {
+                indexArray[i*2] = i * decimationWidth;
+                break;
+            }
+        }
+
+        size_t* minIndex = new size_t[1];
```

Comment:
should just be `size_t minIndex;` and then pass it in as `&minIndex`, otherwise you're leaking memory.